hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientGrpc.java

```diff
@@ -244,8 +248,12 @@ public ContainerCommandResponseProto sendCommand(
       XceiverClientReply reply;
       reply = sendCommandWithTraceIDAndRetry(request, validators);
       return reply.getResponse().get();
-    } catch (ExecutionException | InterruptedException e) {
+    } catch (ExecutionException e) {
       throw new IOException("Failed to execute command " + request, e);
+    } catch (InterruptedException e) {
```

Comment:
Now when Interrupted Exception happens, we return null. But I see in some places callers do not have a check for not null.

For instance
**ContainerProtocolCalls.java L 117:**
```
    ContainerCommandResponseProto response =
        xceiverClient.sendCommand(request, getValidatorList());
```

Reply:
Thank you for catching this @bharatviswa504 .
It seems that in all such callers, the response is assumed to be non-null.

We can address this in following ways:
1. Avoid returning NULL and instead rethrow the InterruptedException
2. Return empty `response` instance
3. Return NULL and put a null check in caller method.

I am more inclined to option 1 as it keeps the code base simple.

What are your thoughts ?

Reply:
Yes, we can rethrow with IOException

Reply:
Updated PR. Thanks @bharatviswa504 !